shell.nix

```
@@ -1,8 +1,4 @@
-{ pkgs ? import (builtins.fetchTarball {
-  url =
-    "https://github.com/NixOS/nixpkgs/archive/cda0e75a0bd7cf05bd3e40658c163e4f8f376b7b.tar.gz";
-  sha256 = "sha256-Toz3HEHeq6Esr5uDOMel8BiGSa94gj+og3Yz4YEgjYI=";
-}) { } }:
+{pkgs ? abort "This project is heavily dependent on flakes. Use `nix develop` instead."}:
```

```suggestion
{ pkgs ? (
    let
      lock = builtins.fromJSON (builtins.readFile ./flake.lock);
    in import (builtins.fetchTarball {
      url = "https://github.com/NixOS/nixpkgs/archive/${lock.nodes.nixpkgs.locked.rev}.tar.gz";
      sha256 = lock.nodes.nixpkgs.locked.narHash;
  }) {}
  )
}:
```

See comment above, also see https://nixos.wiki/wiki/FAQ/Pinning_Nixpkgs for more info

---

Okay this is brilliant. I feel embarassed for some of the previous MRs I had.